### PR TITLE
Adding tests for util/httputil/client

### PR DESCRIPTION
**Pull request by pafuent**
Adding tests for util/httputil/client with a 100% coverage.
Removing the NewDeadlineRoundTripper from util/httputil/client because
is not used.
Adding a new test util to check http.Request in http.RoundTrip interface
implementors.

### CODE REVIEW

**Comment by pafuent** [comment]
@Gouthamve and @matthiasr, Could you take a look at this PR?

**Comment by Gouthamve** [comment]
Hi, sorry for taking so long but the changes look good to me and I think the test cases are solid.

Having said that, I am not really familiar with the code-path and edge cases and I think @brian-brazil is a better person to take a look at this.

File: util/httputil/client_test.go [hunk 1]
```diff
@@ -0,0 +1,412 @@
...
+	ServerCertificatePath = "testdata/server.crt"
+	ServerKeyPath         = "testdata/server.key"
+	BarneyCertificatePath = "testdata/barney.crt"
+	BarneyKeyNoPassPath   = "testdata/barney-no-pass.key"
+	MissingCA             = "missing/ca.crt"
+	MissingCert           = "missing/cert.crt"
+	MissingKey            = "missing/secret.key"
+
+	ExpectedMessage        = "I'm here to serve you!!!"
+	BearerToken            = "meaningoflifeuniverseandeverithingelse"
```
Comment (brian-brazil) [inline util/httputil/client_test.go]: At least get the quote right

File: util/httputil/client_test.go [hunk 2]
```diff
@@ -0,0 +1,412 @@
...
+	if err != nil {
+		t.Fatalf("Can't create a new TLS Config from a configuration (%s).", err)
+	}
+
+	if !reflect.DeepEqual(tlsConfig, expectedTLSConfig) {
+		t.Fatalf("Unexpected TLS Config result: \n\n%+v\n expected\n\n%+v", tlsConfig, expectedTLSConfig)
+	}
+}
+
+var invalidTLSConfig = []struct {
```
Comment (brian-brazil) [inline util/httputil/client_test.go]: These usually go inside the test function.

File: util/httputil/client_test.go (hunk 1 above)
Comment (pafuent) [inline util/httputil/client_test.go]: My bad 😞 (I think this is a clear signal that is time to read it again)

File: util/httputil/client_test.go (hunk 2 above)
Comment (pafuent) [inline util/httputil/client_test.go]: I used https://github.com/prometheus/prometheus/blame/master/config/config_test.go#L577 as inspiration, also is the pattern used in https://golang.org/src/fmt/fmt_test.go#L134. Of course I can change it if you want.

Comment (brian-brazil) [inline util/httputil/client_test.go]: Inside the function is more usual, look further down in config_test for example.

Comment (pafuent) [inline util/httputil/client_test.go]: Ok, I'll change it

**Comment by brian-brazil** [comment]
Thanks!

**Pull request merged**